### ISSUE TITLE
Inter-batch node slicing for model output

### ISSUE THREAD

**Issue by manavkulshrestha** [pyg-team/pytorch_geometric#8236]
### 🚀 The feature, motivation and pitch

Hello, I have a question for how to do something with batching. I have the following Data objects that my dataloader yields (batch size 1): `DataBatch(x=[15, 4], edge_index=[2, 120], y=[5], batch=[15], ptr=[2])`. Now, once my model gives me the output, I only want to get the loss by comparing the node predictions on the first `len(y)` nodes in each graph since those nodes are different and my model output is only defined for them. That is, something like
```
out = model(batch.x, batch.edge_index)
loss = loss_fn(out[:len(batch.y)], batch.y)
```
But, this of course, doesn't really work when we have batching since multiple graph instances are concatenated with each other and the out values I care about are now interleaved. I cannot find any way to achieve what I have described and would like it to be possible.

### Alternatives

Of course, I could train with a batch size of 1, but this is very slow

### Additional context

N/A

**Comment by rusty1s** [comment]
Let's move the discussion to https://github.com/pyg-team/pytorch_geometric/discussions/8235.
